### PR TITLE
Sandbox Juicer

### PR DESCRIPTION
**Pull request by colton-smith**

### CODE REVIEW

File: Sandbox/Entity.cpp [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#include "Entity.h"
+
+// Entity
```
Comment (liamjwarren) [inline Sandbox/Entity.cpp:3]: Line here

File: Sandbox/Entity.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+			m_color
+		);
+	}
+
+	int m_id;
+	glm::vec2 m_position;
+	glm::vec2 m_scale;
+	glm::vec4 m_color;
+	float m_rotation;
+};
```
Comment (liamjwarren) [inline Sandbox/Entity.h]: Line here

**Review by liamjwarren — APPROVED** [review]
This is sweeeeet! I like how it shows/tests so many different things. One thing I noticed was that the imGui menu said there were options to configure under configurations but there were not any.

File: Sandbox/Sandbox.h [hunk 3]
```diff
@@ -21,6 +26,8 @@ class SandboxLayer : public Aquarius::Layer
 	Aquarius::sharedPtr<Aquarius::OrthographicCamera> m_Camera;
 	Aquarius::sharedPtr<Aquarius::Texture> m_texture;
 
+	std::vector<Entity*> m_entities;
```
Comment (Burke-Daniel) [inline Sandbox/Sandbox.h:29]: Unique pointers are cancelled :(((

File: Sandbox/Sandbox.cpp [hunk 4]
```diff
@@ -30,67 +40,196 @@ void SandboxLayer::onCreation()
...
+
+            if (ImGui::Button("Pop Entity"))
+            {
+                // TODO - Make it so you can configure the entity
+                if (m_entities.size() != 0)
+                {
+                    Entity* ent = m_entities.back();
+                    delete ent;
+                    ent = nullptr;
+                    m_entities.pop_back();
```
Comment (Burke-Daniel) [inline Sandbox/Sandbox.cpp]: If you use unique pointer, when you call pop_back, the delete will get called automatically by the destructor of unique pointer

File: Sandbox/Sandbox.cpp [hunk 5]
```diff
@@ -102,13 +241,89 @@ Sandbox::Sandbox()
...
+
+    if (node_open)
+    {
+        // POSITION
+        ImGui::PushID("Position"); // Use field index as identifier.
+        ImGui::TableNextRow();
+        ImGui::TableSetColumnIndex(0);
+
+        // Here we use a TreeNode to highlight on hover (we could use e.g. Selectable as well)
+        ImGuiTreeNodeFlags flags = ImGuiTreeNodeFlags_Leaf | ImGuiTreeNodeFlags_NoTreePushOnOpen | ImGuiTreeNodeFlags_Bullet;
```
Comment (Burke-Daniel) [inline Sandbox/Sandbox.cpp:379]: Oooo this is cool

**Review by Burke-Daniel — APPROVED** [review]
This is honestly sick and you are the designated ImGUI expert now lol. My only comment is me being one of those crazy modern C++ people who frown upon the keywords new and delete lol. But honestly this is GTM either way!

File: Sandbox/Sandbox.h (hunk 3 above)
Comment (liamjwarren) [reply Sandbox/Sandbox.h:29]: #woke

Comment (Burke-Daniel) [reply Sandbox/Sandbox.h:29]: All these zoomers are out here managing their own memory again

**Review by Burke-Daniel — APPROVED** [review]
GTM, full send

**Pull request merged**